nodes/list_masks/mask_join.py

```diff
@@ -87,28 +103,29 @@ def apply_mask(self, mask, data_t, data_f):
         return out
 
     def get_level(self, mask, data_t, data_f, level):
+        if self.choice:
+            apply_mask = self.apply_choice_mask
+        else:
+            apply_mask = self.apply_mask
+            
         if level == 1:
             out = []
-            param = (mask, data_t, data_f)
+            param = list_match_func[self.list_match_global]([mask, data_t, data_f])
             if not all((isinstance(p, (list, tuple)) for p in param)):
                 print("Fail")
                 return
-            max_index = min(map(len, param))
-            if self.choice:
-                apply_mask = self.apply_choice_mask
-            else:
-                apply_mask = self.apply_mask
 
-            for i in range(max_index):
-                out.append(apply_mask(mask[i], data_t[i], data_f[i]))
+            for m, t, f in zip(*param):
+                out.append(apply_mask(m, t, f))
             return out
-        elif level > 2:
+        elif level >= 2:
             out = []
-            for t, f in zip(data_t, data_f):
+            param = list_match_func[self.list_match_global]([data_t, data_f])
+            for t, f in zip(*param):
                 out.append(self.get_level(mask, t, f, level - 1))
             return out
         else:
-            return self.apply_mask(mask[0], data_t, data_f)
+            return apply_mask(mask[0], data_t, data_f)
```

Comment:
Now it's not clear which particular cases this statement should handle.

Reply:
It makes sure that true and false data have the exact same structure at each level, otherwise it is unclear how to switch between the two.

Reply:
I mean

```
level = 1, 2, 3, ..., n

if level == 1:
elif level >= 2:
else:
```
What value should level have to reach the else statement?

Reply:
0

Reply:
```py
level: IntProperty(name="Level", default=1, min=1, update=updateNode)
```

Currently level can't be less than one according to the property declaration and I don't think that 0 level has any sense. Maybe the function should raise an error about unexpected value of given level?

Reply:
No, no, in the `process `function, when calling `get_levels`, it is not `level `property which is passed, but `level - 1`, so everything's fine.